cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -633,6 +748,188 @@ typename DataHelper::NodeType::Base * TrainBatchTaskBase<algorithmFPType, DataHe
     return makeLeaf(_aSample.get() + iStart, n, curImpurity, nClasses);
 }
 
+template <typename algorithmFPType, typename DataHelper, CpuType cpu>
+typename DataHelper::NodeType::Base * TrainBatchTaskBase<algorithmFPType, DataHelper, cpu>::buildBestFirst(
+    services::Status & s, size_t iStart, size_t n, size_t level, typename DataHelper::ImpurityData & curImpurity, bool & bUnorderedFeaturesUsed,
+    size_t nClasses)
+{
+    struct WorkItem
+    {
+        bool isLeaf;
+        bool featureUnordered;
+        size_t start;
+        size_t n;
+        size_t nLeft;
+        size_t level;
+        double improvement;
+        typename DataHelper::ImpurityData impurityLeft;
+        typename DataHelper::ImpurityData impurityRight;
+        typename DataHelper::NodeType::Split * node;
+
+        WorkItem() : isLeaf(true), featureUnordered(false), start(0), n(0), nLeft(0), level(0), improvement(0.0), node(nullptr) {}
+
+        WorkItem(bool featureUnordered, size_t start, size_t n, size_t level)
+            : isLeaf(true), featureUnordered(featureUnordered), start(start), n(n), nLeft(0), level(level), improvement(0.0), node(nullptr)
+        {}
+
+        void copy(WorkItem & src)
```

Comment:
Maybe better make a constructor copy?
```c++
WorkItem(const WorkItem & src)
{
...
}

Reply:
operator=(...) is used instead of copy(...) function